EpiResponsivePicture/TagBuilders/PictureTagBuilder.cs

```diff
@@ -0,0 +1,154 @@
+            return this;
+        }
+
+        public IPictureTagBuilder WithContentReference(ContentReference contentReference)
+        {
+            pictureContentReference = contentReference;
+            return this;
+        }
+
+        public IPictureTagBuilder WithFallbackUrl(string fallbackUrl)
+        {
+            pictureFallbackUrl = fallbackUrl;
+            return this;
+        }
+
+        public IPictureTagBuilder WithViewModel(ResizedPictureViewModel viewModel)
+        {
+            pictureViewModel = viewModel;
+            return this;
+        }
+
+        public TagBuilder Build()
+        {
+            Guard.IsNotNull(pictureProfile, nameof(pictureProfile));
+            
+            pictureFallbackUrl ??= string.Empty;
+            pictureViewModel ??= new ResizedPictureViewModel();
+            
+            GetPictureData();
+
+            var sourceTagBuilder = SourceTagBuilder
+                .Create()
+                .WithImageUrl(pictureUrl)
+                .WithFocalPoint(focalPoint)
+                .WithResizedImageFormat(pictureProfile.Format);
+
+            foreach (var pictureSource in pictureProfile.Sources)
+            {
+                var sourceTag = sourceTagBuilder.WithSource(pictureSource).Build();
+
+                element.InnerHtml.AppendHtml(sourceTag.RenderSelfClosingTag());
+                
+                sourceTagBuilder.Clear();
+            }
+            
+            AddAdditionalAttributes();
+
+            var imgTag = CreateImageElement();
+            element.InnerHtml.AppendHtml(imgTag.RenderSelfClosingTag());
+            
+            return element;
+        }
+        
+        private void GetPictureData()
+        {
+            var imageFound = ServiceLocator.Current.GetInstance<IContentLoader>().TryGet<IContentData>(pictureContentReference,
+                new LoaderOptions { LanguageLoaderOption.FallbackWithMaster() }, out var content) && content is IImage;
+            
+            if(imageFound)
+                GetImgAltText((IImage) content);
+
+            GetFocalPoint(content as IResponsiveImage);
+            SetPictureProfilesAspectRatios(content as IResponsiveImage);
+
+            pictureUrl = imageFound ? ResolveUrl() : pictureFallbackUrl;
+        }
+
+        private void GetImgAltText(IImage image)
+        {
+            altImgText = image.Description ?? string.Empty;
+        }
+        
+        private void GetFocalPoint(IResponsiveImage responsiveImage)
+        {
+            focalPoint = responsiveImage?.FocalPoint ?? FocalPoint.Center;
+        }
+
+        private void SetPictureProfilesAspectRatios(IResponsiveImage responsiveImage)
+        {
+            foreach (var pictureProfileSource in pictureProfile.Sources)
+            {
+                if (responsiveImage is null || pictureProfileSource.TargetAspectRatio is null)
+                {
+                    pictureProfileSource.TargetAspectRatio = AspectRatio.Default; // 4:3 
```

Comment:
Note: hidden bug in here. PictureProfile is being modified while rendering a picture.